src/mariadb/videx/ha_videx.cc

```diff
@@ -0,0 +1,992 @@
+	videx_log_ins.markRecordInRange(FUNC_FILE_LINE, min_key, max_key, key, &request_item);
+
+	std::string n_rows_str;
+	ha_rows n_rows;
+	THD* thd = ha_thd();
+  	int error = ask_from_videx_http(request_item, n_rows_str, thd);
+	if (error) {
+		n_rows = 10; // default number to force index
+	}
+	else {
+		n_rows = std::stoull(n_rows_str);
+	}
+
+	DBUG_RETURN(n_rows);
+}
+
+ha_rows ha_videx::estimate_rows_upper_bound()
+{
+	DBUG_ENTER("ha_videx::estimate_rows_upper_bound");
+	DBUG_RETURN(10);
+}
+
+/** 
+Create a new table to an VIDEX database. Not required for VIDEX. */
+
+int ha_videx::create(const char *name, TABLE *table_arg,
+                       HA_CREATE_INFO *create_info)
+{
+	DBUG_ENTER("ha_videx::create");
+	DBUG_PRINT("info", ("name: %s, table_arg: %p, create_info: %p", name, table_arg, create_info));
+	DBUG_RETURN(0);
+}
+
+int ha_videx::delete_table(const char *name)
+{
+	DBUG_ENTER("ha_videx::delete_table");
+	DBUG_RETURN(0);
+}
+
+int ha_videx::rename_table(const char* from, const char* to)
+{
+	DBUG_ENTER("ha_videx::rename_table");
+	DBUG_RETURN(0);
+}
+
+/**
+DS-MRR implementation.
+*/
+
+int ha_videx::multi_range_read_init(
+	RANGE_SEQ_IF*	seq,
+	void*		seq_init_param,
+	uint		n_ranges,
+	uint		mode,
+	HANDLER_BUFFER*	buf)
+{
+	return(m_ds_mrr.dsmrr_init(this, seq, seq_init_param,
+				 n_ranges, mode, buf));
+}
+
+int ha_videx::multi_range_read_next(
+	range_id_t*		range_info)
+{
+	return(m_ds_mrr.dsmrr_next(range_info));
+}
+
+ha_rows ha_videx::multi_range_read_info_const(
+	uint		keyno,
+	RANGE_SEQ_IF*	seq,
+	void*		seq_init_param,
+	uint		n_ranges,
+	uint*		bufsz,
+	uint*		flags,
+	ha_rows         limit,
+	Cost_estimate*	cost)
+{
+	m_ds_mrr.init(this, table);
+
+	return (m_ds_mrr.dsmrr_info_const(keyno, seq, seq_init_param,
+									   n_ranges,
+									   bufsz, flags, limit, cost));
+}
+
+ha_rows ha_videx::multi_range_read_info(
+	uint		keyno,
+	uint		n_ranges,
+	uint		keys,
+	uint		key_parts,
+	uint*		bufsz,
+	uint*		flags,
+	Cost_estimate*	cost)
+{
+	m_ds_mrr.init(this, table);
+	return (m_ds_mrr.dsmrr_info(keyno, n_ranges, keys, key_parts, bufsz,
+					flags, cost));
+}
+
+int ha_videx::multi_range_read_explain_info(uint mrr_mode,
+	char *str, size_t size)
+{
+	return(m_ds_mrr.dsmrr_explain_info(mrr_mode, str, size));
+}
+
+/** Attempt to push down an index condition.
+@param[in] keyno MySQL key number
+@param[in] idx_cond Index condition to be checked
+@return Part of idx_cond which the handler will not evaluate */
+
+class Item* ha_videx::idx_cond_push(
+	uint		keyno,
+	class Item*	idx_cond)
+{
+	DBUG_ENTER("ha_videx::idx_cond_push");
+	DBUG_ASSERT(keyno != MAX_KEY);
+	DBUG_ASSERT(idx_cond != NULL);
+
+	pushed_idx_cond = idx_cond;
+	pushed_idx_cond_keyno = keyno;
+	in_range_check_pushed_down = TRUE;
+	DBUG_RETURN(NULL);
+}
+
+/**
+Returns table statistics using the VIDEX server (HTTP).
+Populates `stats` and related fields.
+@return 0 on success or HA_ERR_*.
+*/
+
+int ha_videx::info_low(uint flag, bool is_analyze)
```

Comment:
We can simply add `refer to ha_videx::records_in_range` here, avoiding doc duplicate.